addons/web/static/src/views/form/form_controller.js

```diff
@@ -152,10 +152,10 @@ export class FormController extends Component {
 
         // enable the archive feature in Actions menu only if the active field is in the view
         this.archiveEnabled =
-            "active" in activeFields
-                ? !activeFields.active.readonly
-                : "x_active" in activeFields
-                ? !activeFields.x_active.readonly
+            "active" in this.props.fields
+                ? !this.props.fields.active.readonly
+                : "x_active" in this.props.fields
+                ? !this.props.fields.x_active.modifiers.readonly
```

Comment:
It looks like in a form view the archive action should never be there if the "active" field is not written in the arch.
@aab-odoo is it true ?

```suggestion
            "active" in activeFields
                ? !this.props.fields.active.readonly
                : "x_active" in activeFields
                ? !this.props.fields.x_active.readonly
```

Reply:
Ho indeed. Otherwise we can't know whether we must display "Archive" or "Unarchive"

Reply:
Just to make things clear, in list views we do not account for the `active` field being present in their archs. => expected behavior, isn't it ?

Reply:
Is it well done now ?

Reply:
yes (y)